packages/search-in-workspace/src/common/search-in-workspace-interface.ts

```diff
@@ -53,6 +53,10 @@ export interface SearchInWorkspaceOptions {
      * Glob pattern for matching files and directories to exclude the search.
      */
     exclude?: string[];
+    /**
+     * Glob pattern to match file and directories based on the preferences 'files.exclude' and 'search.excludeFiles'.
+     */
+    excludedGlobs?: string[]
```

Comment:
we use `exclude` already for this.

Reply:
You aslo leaking frontend details here. Backend API does not need to know from there exclude are coming.

Reply:
Thank you for your review, @akosyakov.

> we use `exclude` already for this.

I believe the `exclude` we used was for the `files to exclude` input field, but looking at the interface defined in vscode, they only have `excludePattern` as one of its properties. Would it be better to integrate the `files.exclude` and `search.excludeFiles` defaults in the `exclude` rather than creating a new property?

 

> You aslo leaking frontend details here. Backend API does not need to know from there exclude are coming.

Yes, you're right. I can update the description to not include frontend details over here.